metadata/src/main/java/org/apache/kafka/controller/AclControlManager.java

```diff
@@ -256,11 +224,6 @@ public void replay(
             throw new RuntimeException("Unable to replay " + record + " for " + acl +
                 ": acl not found " + "in existingAcls.");
         }
-        if (!snapshotId.isPresent()) {
```

Comment:
Nice to see this code going away. It always tripped me up 😅

Reply:
Yeah, it was always a messy corner case.